libminifi/include/core/logging/Logger.h

```diff
@@ -59,11 +59,18 @@ inline char const* conditional_conversion(const utils::SmallString<N>& arr) {
   return arr.c_str();
 }
 
-template<typename T>
+template<typename T, typename = typename std::enable_if<
+    std::is_arithmetic<T>::value ||
+    std::is_enum<T>::value ||
+    std::is_pointer<T>::value>::type>
 inline T conditional_conversion(T const& t) {
   return t;
 }
 
+inline char const* conditional_conversion(const char* str) {
+  return str;
+}
```

Comment:
for some reason I cannot reply to your comment about `std::hash` specialization:
from [google guidelines](https://google.github.io/styleguide/cppguide.html#std_hash)

> Do not define specializations of std::hash.

Reply:
for this comment, it is needed so we can allow [this](https://godbolt.org/z/9drc8f) to compile, and now that I wrote this down, I realize I could simply remove the `const&` qualifier on the template

```
#include <iostream>

struct A{
  static constexpr const char* STR = "abcd";
};

template<typename T, typename = typename std::enable_if<
    std::is_arithmetic<T>::value ||
    std::is_enum<T>::value ||
    std::is_pointer<T>::value>::type>
inline T conditional_conversion(T const& t) {
  return t;
}

// inline char const* conditional_conversion(const char* str) {
//   return str;
// }

int main(){
  std::cout << conditional_conversion(A::STR) << std::endl;
}
```

Reply:
The reason you couldn't reply to my comment about hash is because it was a reply to this discussion: https://github.com/apache/nifi-minifi-cpp/pull/932#discussion_r513307831

To this thread: acknowledged, I'm fine with either way.